apps/ex_wire/test/integration/node_discovery_test.exs

```diff
@@ -26,31 +32,59 @@ defmodule ExWire.NodeDiscoveryTest do
         kademlia_process_name: kademlia_process_name,
         network_adapter: network_adapter,
         supervisor_name: supervisor_name,
-        port: port
+        port: port,
+        nodes: @bootnodes
       ]
     })
 
     {:ok,
      %{
        kademlia_name: kademlia_process_name,
-       udp_name: udp_name,
-       supervisor_name: supervisor_name
+       supervisor_name: supervisor_name,
+       udp_name: udp_name
      }}
   end
 
   test "receives pong from remote node and adds it to local routing table", %{
+    kademlia_name: kademlia_name
+  } do
+    expected_node = expected_node()
+    Kademlia.ping(expected_node, process_name: kademlia_name)
+
+    Process.sleep(1_000)
+
+    routing_table = Kademlia.routing_table(process_name: kademlia_name)
+    assert RoutingTable.member?(routing_table, expected_node())
+  end
+
+  test "request neighbours from remote node and pings received nodes", %{
     kademlia_name: kademlia_name,
     udp_name: udp_name
   } do
-    remote_endpoint = remote_endpoint()
-    current_endpoint = current_endpoint()
-    ping = Ping.new(current_endpoint, remote_endpoint)
+    find_neighbours = FindNeighbours.new(Config.node_id())
 
-    Network.send(ping, udp_name, remote_endpoint)
-    Process.sleep(1000)
+    Network.send(find_neighbours, udp_name, remote_endpoint())
+
+    Process.sleep(2_000)
 
     routing_table = Kademlia.routing_table(process_name: kademlia_name)
-    assert RoutingTable.member?(routing_table, expected_node())
+    pings_count = routing_table.expected_pongs |> Map.values() |> Enum.count()
+
+    assert pings_count > 0
+  end
+
+  test "successfully finished discovery and adds nodes to routing_table", %{
+    kademlia_name: kademlia_name
+  } do
+    # let's wait for all discovery rounds
+    Process.sleep(30_000)
```

Comment:
Should we tag these tests so that we don't run them all the time since they're sleeping? We should run them on CI in a full run, but maybe we can exclude them by default. What do you think?

Reply:
@vyorkin yes, it is tagged